src/app/(proper_react)/redesign/(authenticated)/user/(dashboard)/dashboard/View.tsx

```diff
@@ -252,20 +256,54 @@ export const View = (props: Props) => {
     </>
   );
 
+  const waitlistTriggerRef = useRef<HTMLAnchorElement>(null);
+  const dialogTriggerState = useOverlayTriggerState({});
+  const overlayTrigger = useOverlayTrigger(
+    { type: "dialog" },
+    dialogTriggerState,
+    waitlistTriggerRef,
+  );
+
   const freeScanCta = props.isEligibleForFreeScan && (
-    <p>
-      {l10n.getFragment("dashboard-exposures-all-fixed-free-scan", {
-        vars: {
-          data_broker_total_num: parseInt(
-            process.env.NEXT_PUBLIC_ONEREP_DATA_BROKER_COUNT as string,
-            10,
-          ),
-        },
-        elems: {
-          a: <a href="/redesign/user/welcome/free-scan?referrer=dashboard" />,
-        },
-      })}
-    </p>
+    <>
+      <SubscriberWaitlistDialog
+        triggerRef={waitlistTriggerRef}
+        dialogTriggerState={dialogTriggerState}
+        overlayTrigger={overlayTrigger}
+      />
+      <p>
+        {l10n.getFragment("dashboard-exposures-all-fixed-free-scan", {
+          vars: {
+            data_broker_total_num: parseInt(
+              process.env.NEXT_PUBLIC_ONEREP_DATA_BROKER_COUNT as string,
+              10,
+            ),
+          },
+          elems: {
+            a:
+              props.totalNumberOfPerformedScans <
+              parseInt(
+                process.env.NEXT_PUBLIC_ONEREP_MAX_SCANS_THRESHOLD as string,
+                10,
+              ) ? (
+                <a
+                  ref={waitlistTriggerRef}
+                  href="/redesign/user/welcome/free-scan?referrer=dashboard"
+                />
+              ) : (
+                <Button
+                  variant="ternary"
+                  buttonRef={waitlistTriggerRef}
+                  {...overlayTrigger.triggerProps}
+                  onPress={() => {
+                    dialogTriggerState.open();
+                  }}
+                />
```

Comment:
We can’t pass the entire `SubscriberWaitlistDialog` component here. So in this case we only provide the trigger and pass the relevant props to `SubscriberWaitlistDialog`.